martian-derive/src/lib.rs

```diff
@@ -99,7 +99,7 @@ pub fn make_mro(
             Volatile::Strict => quote![volatile: Some(::martian::Volatile::Strict),],
             Volatile::False => quote![volatile: Some(::martian::Volatile::False),],
         },
-        None => quote![volatile: Some(::martian::Volatile::default()),],
+        None => quote![volatile: None,],
```

Comment:
`None` is very much not the same as `false`.  However I don't think it's a good idea to change this back.  Until such time as the `mrp` bug with VDR is fixed, it's better to just `retain` whatever is required but keep the benefits of `strict` mode otherwise.  Because there is other surprising and buggy behavior with non-strict mode.

Reply:
https://github.com/martian-lang/martian-rust/pull/249/files#diff-30338e3e5dc4137e7cac862e794bb35cd61efd112945e252035d3939d71a75f4R102
I'm reverting this change in this PR https://github.com/martian-lang/martian-rust/pull/249 that changed the default value from `None` to `Some(Volatile::default())` which that PR changed to `strict`.
- https://github.com/martian-lang/martian-rust/pull/249

Reply:
Do you mean that Martian treats `volatile` omitted as different than `volatile = false`?

Reply:
> Until such time as the mrp bug with VDR is fixed, it's better to just retain whatever is required but keep the benefits of strict mode otherwise. Because there is other surprising and buggy behavior with non-strict mode.

Excellent if you're volunteering to do that. In the mean time, the two PRs
- https://github.com/10XDev/cellranger/pull/11024
- https://github.com/10XDev/cellranger/pull/11020

are both blocked from being merged, which is not a good current state of affairs. I'd like to merge this PR and the two `cellranger` PRs.

Reply:
1. Yes, I do mean that `volatile` being omitted is not the same as `volatile = false`.  `volatile = false` means completely disable VDR for stage.  Omitting it means use the default mode based on `--vdrmode`.  Specifically:
    - use `strict` if `--vdrmode=strict`
    - run VDR when all dependent stages have completed if _called_ with `volatile = true`
    - Otherwise, delete all chunk files immediately but don't delete any split or join files (which is pretty weird behavior if you ask me).
2. What I meant by the other comment is that the preferred solution (that I have time for) is to use `volatile = strict` but to `retain` outputs relevant to the martian VDR bug.

My objection to reverting the change in default is that there's a lot of stages which _should_ have `volatile = strict` but don't, and having that not be the default (and instead have that weird thing be the default) is unfortunate behavior.